receiver/prometheusreceiver/internal/prom_to_otlp.go

```diff
@@ -54,7 +56,49 @@ func CreateNodeAndResourcePdata(job, instance, scheme string) *pcommon.Resource
 	}
 	attrs.UpsertString(conventions.AttributeServiceInstanceID, instance)
 	attrs.UpsertString(conventions.AttributeNetHostPort, port)
-	attrs.UpsertString(conventions.AttributeHTTPScheme, scheme)
+	attrs.UpsertString(conventions.AttributeHTTPScheme, serviceDiscoveryLabels.Get(model.SchemeLabel))
+
+	addKubernetesResource(attrs, serviceDiscoveryLabels)
 
 	return &resource
 }
+
+func addKubernetesResource(attrs pcommon.Map, serviceDiscoveryLabels labels.Labels) {
+	if podName := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_name"); podName != "" {
+		attrs.UpsertString(conventions.AttributeK8SPodName, podName)
+	}
+	if podUID := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_uid"); podUID != "" {
+		attrs.UpsertString(conventions.AttributeK8SPodUID, podUID)
+	}
+	if containerName := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_container_name"); containerName != "" {
+		attrs.UpsertString(conventions.AttributeK8SContainerName, containerName)
+	}
+	if nodeName := serviceDiscoveryLabels.Get("__meta_kubernetes_pod_node_name"); nodeName != "" {
+		attrs.UpsertString(conventions.AttributeK8SNodeName, nodeName)
+	}
+	if nodeName := serviceDiscoveryLabels.Get("__meta_kubernetes_node_name"); nodeName != "" {
+		attrs.UpsertString(conventions.AttributeK8SNodeName, nodeName)
+	}
+	if nodeName := serviceDiscoveryLabels.Get("__meta_kubernetes_endpoint_node_name"); nodeName != "" {
+		attrs.UpsertString(conventions.AttributeK8SNodeName, nodeName)
+	}
```

Comment:
thought: should this just be a map from `__meta*` name to `conventions.*` so that there's less duplication and makes it easier to add another mapping in the future?

Reply:
done